playback/src/decoder/symphonia_decoder.rs

```diff
@@ -131,6 +187,71 @@ impl SymphoniaDecoder {
         }
     }
 
+    pub(crate) fn local_file_metadata(&mut self) -> Option<LocalFileMetadata> {
+        let metadata = self.metadata()?;
+        let tags = metadata.current()?.tags();
+        let mut metadata = LocalFileMetadata::default();
+
+        for tag in tags {
+            if let Value::String(value) = &tag.value {
+                match tag.std_key {
+                    // We could possibly use mem::take here to avoid cloning, but that risks leaving
+                    // the audio item metadata in a bad state.
+                    Some(StandardTagKey::TrackTitle) => metadata.name = value.clone(),
+                    Some(StandardTagKey::Language) => metadata.language = value.clone(),
+                    Some(StandardTagKey::Artist) => metadata.artists = value.clone(),
+                    Some(StandardTagKey::AlbumArtist) => metadata.album_artists = value.clone(),
+                    Some(StandardTagKey::Album) => metadata.album = value.clone(),
+                    Some(StandardTagKey::TrackNumber) => {
+                        metadata.number = value.parse::<u32>().unwrap_or_default()
+                    }
+                    Some(StandardTagKey::DiscNumber) => {
+                        metadata.disc_number = value.parse::<u32>().unwrap_or_default()
+                    }
+                    _ => (),
+                }
+            } else if let Value::UnsignedInt(value) = &tag.value {
+                match tag.std_key {
+                    Some(StandardTagKey::TrackNumber) => metadata.number = *value as u32,
+                    Some(StandardTagKey::DiscNumber) => metadata.disc_number = *value as u32,
+                    _ => (),
+                }
+            } else if let Value::SignedInt(value) = &tag.value {
+                match tag.std_key {
+                    Some(StandardTagKey::TrackNumber) => metadata.number = *value as u32,
+                    Some(StandardTagKey::DiscNumber) => metadata.disc_number = *value as u32,
+                    _ => (),
+                }
+            }
+        }
+
+        Some(metadata)
+    }
+
+    fn metadata(&mut self) -> Option<Metadata<'_>> {
+        let mut metadata = self.format.metadata();
+
+        // If we can't get metadata from the container, fall back to other tags found by probing.
+        // Note that this is only relevant for local files.
+        if metadata.current().is_none() {
+            if let Some(ref mut probe_metadata) = self.probed_metadata {
+                if let Some(inner_probe_metadata) = probe_metadata.get() {
+                    metadata = inner_probe_metadata;
+                }
+            }
+        }
```

Comment:
This part is new, I found calling `self.format.metadata` didn't really work with the files I had, and found this snippet in Symphonia's player example: https://github.com/pdeljanov/Symphonia/blob/master/symphonia-play/src/main.rs#L533-L536

Reply:
Yep that makes sense: sometimes the metadata is tagged onto the track, sometimes it's tagged onto the container (e.g. Ogg is the container and Vorbis is the track; or MP4 is the container and FLAC is the track, and so on).

However take care to also forward to `.current()` so you get the latest version of the metadata fields, in case there are multiple versions of it.

Reply:
Thanks for clarifying - I could do with knowing more about audio formats than I do right now :)

> However take care to also forward to `.current()` so you get the latest version of the metadata fields

I might be missing something but I think we already are? We reassign `metadata` (and I assume `metadata.current()` by extension) which then gets looped through to ensure we have the latest revision. Is there something else specific that I need to do?

Reply:
Yup all good - I looked to quickly.